src/wp-includes/block-supports/duotone.php

```diff
@@ -352,55 +352,52 @@ function wp_tinycolor_string_to_rgb( $color_str ) {
 	}
 }
 
-
 /**
- * Registers the style and colors block attributes for block types that support it.
+ * Returns the prefixed id for the duotone filter for use as a CSS id.
  *
- * @since 5.8.0
+ * @since 5.9.0
  * @access private
  *
- * @param WP_Block_Type $block_type Block Type.
+ * @param  array $preset Duotone preset value as seen in theme.json.
+ * @return string        Duotone filter CSS id.
  */
-function wp_register_duotone_support( $block_type ) {
-	$has_duotone_support = false;
-	if ( property_exists( $block_type, 'supports' ) ) {
-		$has_duotone_support = _wp_array_get( $block_type->supports, array( 'color', '__experimentalDuotone' ), false );
-	}
-
-	if ( $has_duotone_support ) {
-		if ( ! $block_type->attributes ) {
-			$block_type->attributes = array();
-		}
-
-		if ( ! array_key_exists( 'style', $block_type->attributes ) ) {
-			$block_type->attributes['style'] = array(
-				'type' => 'object',
-			);
-		}
-	}
+function wp_get_duotone_filter_id( $preset ) {
+	return 'wp-duotone-' . $preset['slug'];
 }
 
 /**
- * Renders the duotone filter SVG and returns the CSS filter property to
- * reference the rendered SVG.
+ * Returns the CSS filter property url to reference the rendered SVG.
  *
  * @since 5.9.0
  * @access private
+ *
+ * @param  array $preset Duotone preset value as seen in theme.json.
+ * @return string        Duotone CSS filter property url value.
+ */
+function wp_get_duotone_filter_property( $preset ) {
+	$filter_id = wp_get_duotone_filter_id( $preset );
+	return "url('#" . $filter_id . "')";
+}
 
- * @param array $preset Duotone preset value as seen in theme.json.
- * @return string Duotone CSS filter property.
+/**
+ * Returns the duotone filter SVG string for the preset.
+ *
+ * @since 5.9.0
+ * @access private
+ *
+ * @param  array $preset Duotone preset value as seen in theme.json.
+ * @return string        Duotone SVG filter.
  */
-function wp_render_duotone_filter_preset( $preset ) {
-	$duotone_id     = $preset['slug'];
-	$duotone_colors = $preset['colors'];
-	$filter_id      = 'wp-duotone-' . $duotone_id;
+function wp_get_duotone_filter_svg( $preset ) {
+	$filter_id = wp_get_duotone_filter_id( $preset );
+
 	$duotone_values = array(
 		'r' => array(),
 		'g' => array(),
 		'b' => array(),
 		'a' => array(),
 	);
-	foreach ( $duotone_colors as $color_str ) {
+	foreach ( $preset['colors'] as $color_str ) {
```

Comment:
Will `'colors'` key always exist? If no, then a guard clause is needed for malformed data.

What should happen if it doesn't exist?

One approach is to initialize it:
```suggestion
	if ( ! isset( $preset['colors'] || ! is_array( $preset['colors'] ) ) {
		$preset['colors'] = array();
	}
	
	foreach ( $preset['colors'] as $color_str ) {
```